apps/wakunode2/app.nim

```diff
@@ -596,13 +596,34 @@ proc startRestServer(app: App, address: ValidIpAddress, port: Port, conf: WakuNo
     rest_legacy_filter_api.installLegacyFilterRestApiHandlers(server.router, app.node, legacyFilterCache)
 
     let filterCache = rest_filter_api.MessageCache.init()
-    rest_filter_api.installFilterRestApiHandlers(server.router, app.node, filterCache)
+
+    let filterDiscoHandler = 
+      if app.wakuDiscv5.isSome():
+        some(defaultDiscoveryHandler(app.wakuDiscv5.get(), Filter))
+      else: none(DiscoveryHandler)
```

Comment:
For me it would be strange, that my node stops upon a rest API call. Instead a proper error response would help me in next steps...
Like a /admin call setting up new peers or any other future configuration option.
For example I can imagine that user can change discovery strategies on the go. WDYT?

Reply:
> Instead of returning an empty (none) discovery handler, should we return a default discovery service?

I most cases light nodes won't have discv5 enabled, only service peers and full nodes may have service peers and most would have discv5. What should the default discovery service do?

> If this is not possible, I think we should end the execution at this point and indicate that a proper disco handler should be set. Better to do that than later on responding to the REST clients that "there is no disc service configured" or the current message that is being returned

If we were dealing with only full nodes I would agree but light nodes don't have discovery.

Reply:
> For me it would be strange, that my node stops upon a rest API call. Instead a proper error response would help me in next steps... Like a /admin call setting up new peers or any other future configuration option. For example I can imagine that user can change discovery strategies on the go. WDYT?

You can't enable discv5 at run-time BUT service peers could be added via the Admin API maybe?

Reply:
> > For me it would be strange, that my node stops upon a rest API call. Instead a proper error response would help me in next steps... Like a /admin call setting up new peers or any other future configuration option. For example I can imagine that user can change discovery strategies on the go. WDYT?
> 
> You can't enable discv5 at run-time BUT service peers could be added via the Admin API maybe?

I did not meant for now, but for the future or to allow app writer to provide their own discovery strategy maybe chain them or even dynamically change it via rest interface for ex.
But it was just an example of why might be a better idea to return meaningful error message rather than stopping the node.

Reply:
> For me it would be strange, that my node stops upon a rest API call. Instead a proper error response would help me in next steps... Like a /admin call setting up new peers or any other future configuration option. For example I can imagine that user can change discovery strategies on the go. WDYT?

Afternoon Zoltán! I didn't mean to end due to a REST request. I meant to make the `wakunode2` app to end immediately if no disco handler is given when it starts up. However, to me the best option would be to always have a disco handler.

Reply:
> I most cases light nodes won't have discv5 enabled, only service peers and full nodes may have service peers and most would have discv5. What should the default discovery service do?

> If we were dealing with only full nodes I would agree but light nodes don't have discovery.

Afternoon SP! Good point! Will we have light nodes with rest enabled? I don't think so. In that case, we may always have a default discovery service.

By a "default discovery service" I mean a naïve one which just returns a bunch of hard-coded peers, if possible.

Reply:
> Afternoon SP! Good point! Will we have light nodes with rest enabled? I don't think so. In that case, we may always have a default discovery service.

I don't think we should decide that, but instead leave this open to app devs.

> By a "default discovery service" I mean a naïve one which just returns a bunch of hard-coded peers, if possible.

Infra nodes? I don't think that's a good idea. It clash a bit with the decentralized ethos. We already rely way too much on Infra IMO.

Reply:
Ok perfect, you convinced me that is not a good idea to have a default "discovery service" :)